interp/run.go

```diff
@@ -2036,10 +2034,16 @@ func recv(n *node) {
 	if n.fnext != nil {
 		fnext := getExec(n.fnext)
 		n.exec = func(f *frame) bltn {
-			chosen, v, _ := reflect.Select([]reflect.SelectCase{
-				{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(f.done)},
-				{Dir: reflect.SelectRecv, Chan: value(f)},
-			})
+			ch := value(f)
+			// Fast: channel read doesn't block
+			if x, ok := ch.TryRecv(); ok {
+				if x.Bool() {
```

Comment:
Curious. How do we know `x` represents a `bool`? Shouldn't this be.
```
if x, ok := ch.TryRecv(); ok {
    return tnext
}
```
and falling through in the failing case?

Reply:
When`fnext`  is not `nil`, we are in presence of a  conditional branch (to `tnext` of `fnext`), which occurs only in case of boolean value. This has been detected and set during CFG.  We have to test the content of `x` which contains the boolean value. `ok` only carries if the chan read completed, not the data itself.

Reply:
OK